Lib/site.py

```diff
@@ -234,6 +234,27 @@ def check_enableusersite():
 
     return True
 
+
+def _getuserbase():
+    # Stripped version of sysconfig._getuserbase()
```

Comment:
Can you please elaborate this comment? Please explain that the function was duplicated to speedup Python startup and avoid the sysconfig import in the site module. Add a reference to the bpo 29585.

It seems like you modified this change: you moved "if env_base: return env_base" at the top. Please modify also sysconfig._getuserbase(). I would also prefer the sysconfig also uses sys._framework, instead of get_config_var("PYTHONFRAMEWORK").